src/libraries/System.Private.CoreLib/gen/EventSourceGenerator.Parser.cs

```diff
@@ -3,193 +3,147 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 using System.Threading;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
 
 namespace Generators
 {
     public partial class EventSourceGenerator
     {
-        private sealed class Parser
+        private static bool IsSyntaxTargetForGeneration(SyntaxNode node, CancellationToken cancellationToken) =>
+            node is ClassDeclarationSyntax { AttributeLists.Count: > 0 };
+
+        private static EventSourceClass? GetSemanticTargetForGeneration(GeneratorSyntaxContext context, CancellationToken cancellationToken)
         {
-            private readonly CancellationToken _cancellationToken;
-            private readonly Compilation _compilation;
-            private readonly Action<Diagnostic> _reportDiagnostic;
+            const string EventSourceAutoGenerateAttribute = "System.Diagnostics.Tracing.EventSourceAutoGenerateAttribute";
+            const string EventSourceAttribute = "System.Diagnostics.Tracing.EventSourceAttribute";
 
-            public Parser(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
-            {
-                _compilation = compilation;
-                _cancellationToken = cancellationToken;
-                _reportDiagnostic = reportDiagnostic;
-            }
+            var classDef = (ClassDeclarationSyntax)context.Node;
+            SemanticModel sm = context.SemanticModel;
+            EventSourceClass? eventSourceClass = null;
 
-            public EventSourceClass[] GetEventSourceClasses(List<ClassDeclarationSyntax> classDeclarations)
+            bool autoGenerate = false;
+            foreach (AttributeListSyntax cal in classDef.AttributeLists)
             {
-                INamedTypeSymbol? autogenerateAttribute = _compilation.GetBestTypeByMetadataName("System.Diagnostics.Tracing.EventSourceAutoGenerateAttribute");
-                if (autogenerateAttribute is null)
+                foreach (AttributeSyntax ca in cal.Attributes)
                 {
-                    // No EventSourceAutoGenerateAttribute
-                    return Array.Empty<EventSourceClass>();
-                }
+                    if (sm.GetSymbolInfo(ca, cancellationToken).Symbol is not IMethodSymbol caSymbol)
+                    {
+                        // badly formed attribute definition, or not the right attribute
+                        continue;
+                    }
 
-                INamedTypeSymbol? eventSourceAttribute = _compilation.GetBestTypeByMetadataName("System.Diagnostics.Tracing.EventSourceAttribute");
```

Comment:
Why are we removing the use of GetBestTypeByMetadataName?  Is this different from the other conversation on the regex generator, where we opted to keep its usage?

Reply:
This PR was opened before that conversation. I will revert back to `GetBestTypeByMetadataName` soon.

Reply:
Ok, thanks.